Shopify/theme-check
# Remove the `data/shopify_liquid/(tags|filters|objects).yml` files

### WHY are these changes introduced?

Fixes #656 

### WHAT is this pull request doing?

- Removes the usage of `data/shopify_liquid/tags.yml`, `data/shopify_liquid/filters.yml`, `data/shopify_liquid/objects.yml` in favor of the `SourceIndex`, which holds the most updated values for objects, filters, and tags.
- Transforms remaining `.yml` files to `.json` files, so that we only have 1 format.
- Adds some logic for existing objects, filters, tags, so that it resembles the existing `.yml` files content.

### How to test your changes?

- Verify that code completion for tags is still working as before,
  - and that [the tags mentioned in `LABELS_NOT_IN_SOURCE_INDEX`](https://github.com/Shopify/theme-check/blob/fix-656/lib/theme_check/shopify_liquid/tag.rb#L9) are now suggested too.
- Verify that code completion for objects is still working as before,
  - and that `customer_address` and `product_variant` are suggested now, but offer no further suggestions like attribute suggestions.
- Verify that code completion for filters is still working as before,
  - and that [the filters mentioned in `LABELS_NOT_IN_SOURCE_INDEX`](https://github.com/Shopify/theme-check/blob/fix-656/lib/theme_check/shopify_liquid/filter.rb#L9) are now suggested too.
